Model/Data/OrderUpdate.php

```diff
@@ -196,4 +202,34 @@ private function getOrderUrl(OrderInterface $order): string
     {
         return $this->storeManager->getStore()->getBaseUrl() . 'sales/order/view/order_id/' . $order->getEntityId();
     }
+
+    /**
+     * Validates the order data.
+     *
+     * @param Order $order
+     * @param DataObject $amwalOrderData
+     * @return bool|string True if validation passes, otherwise returns error message.
+     */
+    private function dataValidation(Order $order, DataObject $amwalOrderData)
+    {
+        // Define an associative array mapping Amwal order fields to their corresponding getter methods.
+        // This array is used for comparing Order object values with Amwal order data.
+        $fields = self::FIELD_MAPPINGS;
+        try {
+            if (floatval($order->getBaseGrandTotal()) != floatval($amwalOrderData->getAmount())) {
+                throw new \Exception(sprintf('Order (%s) %s does not match Amwal Order %s (%s != %s)', $order->getIncrementId(), 'base_grand_total', 'amount', $order->getBaseGrandTotal(), $amwalOrderData->getAmount()));
+            }
+            foreach ($fields as $orderMethod => $amwalMethod) {
```

Comment:
You could use the constant het directly instead of singing it to its own variable first